### ISSUE TITLE
dolt table import -u fails to properly parse order of columns

### ISSUE THREAD

**Issue by tommasogritti** [dolthub/dolt#6675]
I am working on a fork of doltcli to ensure it works with the latest dolt.
When updating to dolt 1.15.0, the `test_update_rows` test fails at the [write_rows line](https://github.com/dolthub/doltcli/blob/f4e7c4c453aec34b0e1e45ea7ec19753d8576137/tests/test_write.py#L44).

Digging into the issue, I found the reason is due to the way `dolt table import -u` handles csv files with headers ordered in different ways.

In short:
- `dolt table import -u` WORKS
   - if called with a csv containing **data which results in an update of any of the rows**  AND the **order of the headers in the csv is the same** as the order in the existing table
   - if called with a csv containing **data which results in an addition of new rows** regardless the order of the headers in the csv
- `dolt table import -u` FAILS
   - if called with a csv containing **data which results in an update of any of the rows**  AND the **order of the headers in the csv is different** from the order in the existing table
  
Here is a simple way to reproduce the issue.

Pre-requisites: install [dolt 1.15.0](https://github.com/dolthub/dolt/releases/tag/v1.15.0)

in a bash command line:

```
mkdir test
cd test

dolt init

# create test csv
printf "adjective,date_of_death,id,name\ntragic,1877-01-01,1,Anna\nhonorable,,2,Vronksy\nbuffoon,,3,Oblonksy" > 
test.csv
# import to characters table
dolt table import characters -c --pk=id test.csv

# create update table with same order of the csv headers
printf "adjective,date_of_death,id,name\ntragic,1877-01-01,1,Cleopatra" > update_same_headers_order.csv
# the following works
dolt table import characters -u --pk=id update_same_headers_order.csv

# create another update of id 1, this time with a different oder of the csv headers
printf "date_of_death,id,name,adjective\n1877-01-01,1,Napoleon,tragic" > update_different_headers_order.csv
# the following DOES NOT WORK, resulting in
:<<'COMMENT'
An error occurred while moving data
cause: Incorrect datetime value: 'tragic'
Errors during import can be ignored using '--continue'
COMMENT

dolt table import characters -u --pk=id update_different_headers_order.csv

# create yet another table, again with different order of the csv headers, but this time resulting not in an update but an addition to the table
printf "date_of_death,id,name,adjective\n2000-01-01,4,Millenium,surreal" > add_different_headers_order.csv
# the following works as expected:
dolt table import characters -u --pk=id add_different_headers_order.csv
```

**Comment by timsehn** [comment]
Thanks for the report and detailed repro steps. We will get on this today.

**Comment by max-hoffman** [comment]
This one took me awhile to understand, but I think I'm on track now and ironing out the last few edge cases. Should have a fix this afternoon.

**Comment by max-hoffman** [comment]
This PR https://github.com/dolthub/go-mysql-server/pull/2012 fixes this for me in testing. This was a good bug, on duplicate expressions never behaved correctly for row generator sources when columns were out of order.

**Comment by tommasogritti** [comment]
Thanks for picking this up so quickly @max-hoffman ! 👏

**Comment by max-hoffman** [comment]
@tommasogritti The fix for this should be in release `1.16.0`. Let us know if you see any other issues here, I think we will be giving the INSERT code that import depends on a bit more attention over the next quarter to try to smooth out the experience here.
